crowbar_framework/app/controllers/barclamp_controller.rb

```diff
@@ -270,103 +251,227 @@ def get_proposals_from_barclamps(barclamps)
+  def proposal_show
+    code, message = @service_object.proposal_show(
+      params[:id]
+    )
+
+    raise Crowbar::Error::NotFound if code == 404
+    respond_to do |format|
+      case code
+      when 200
+        format.json do
+          render json: message.raw_data
+        end
+        format.html do
+          @proposal = message
+
+          @active = begin
+            RoleObject.active(
+              params[:controller],
+              params[:id]
+            ).length > 0
+          rescue
+            false
+          end
+
+          flash.now[:alert] = @proposal.fail_reason if @proposal.failed?
+        end
+      else
+        format.json do
+          render json: { error: message }, status: code
+        end
+        format.html do
+          flash[:alert] = message
+
+          redirect_to(
+            root_url
+          )
+        end
       end
-      render inline: {proposals: proposals, i18n: i18n, count: proposals.length}.to_json, cache: false
-    rescue StandardError => e
-      count = (e.class.to_s == "Errno::ECONNREFUSED" ? -2 : -1)
-      lines = ["Failed to iterate over proposal list due to '#{e.message}'"] + e.backtrace
-      Rails.logger.fatal(lines.join("\n"))
-      render inline: {proposals: proposals, count: count, error: e.message}.to_json, cache: false
     end
   end
 
   #
-  # Provides the restful api call for
-  # Create Proposal Instance 	/crowbar/<barclamp-name>/<version>/proposals 	PUT 	Putting a json document will create a proposal
+  # Delete proposal
+  # /crowbar/<barclamp-name>/<version>/proposals/<barclamp-instance-name> DELETE This will remove a proposal
   #
-  add_help(:proposal_create,[:name],[:put])
-  def proposal_create
-    Rails.logger.info "Proposal Create starting. Params #{params.inspect}"
-    controller = params[:controller]
-    orig_id = params[:name] || params[:id]
-    params[:id] = orig_id
-    answer = [500, "Server issue"]
-    begin
-      Rails.logger.info "asking for proposal of: #{params.inspect}"
-      answer = @service_object.proposal_create params
-      Rails.logger.info "proposal is: #{answer.inspect}"
-      unless answer[0] == 200
-        flash[:alert] = answer[1]
+  add_help(:proposal_delete, [:id], [:delete])
+  def proposal_delete
+    code, message = @service_object.proposal_delete(
+      params[:id]
+    )
+
+    raise Crowbar::Error::NotFound if code == 404
+    respond_to do |format|
+      case code
+      when 200
+        format.json do
+          head :ok
+        end
+      else
+        format.json do
+          render json: { error: message }, status: code
+        end
       end
-    rescue StandardError => e
-      flash_and_log_exception(e)
     end
+  end
+
+  #
+  # Commit proposal
+  # /crowbar/<barclamp-name>/<version>/proposals/commit/<barclamp-instance-name> POST This action will create a new instance based upon this proposal. If the instance already exists, it will be edited and replaced
```

Comment:
Metrics/LineLength: Line is too long. [213/150] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)